lib/rubocop/cop/rspec/capybara/visibility_matcher.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Checks for boolean visibility in capybara finders.
+        #
+        # Capybara lets you find elements that match a certain visibiity using
+        # the `:visible` option. `:visible` accepts both boolean and symbols as
+        # values, however using booleans can have unwanted effects. `visible:
+        # false` does not find just invisible elements, but both visible and
+        # invisible elements. For expressiveness and clarity, use one of the
+        # symbol values, `:all`, `:hidden` or `:visible`.
+        # (https://www.rubydoc.info/gems/capybara/Capybara%2FNode%2FFinders:all)
+        #
+        # @example
+        #
+        #   # bad
+        #   expect(page).to have_selector('.foo', visible: false)
+        #
+        #   # bad
+        #   expect(page).to have_selector('.foo', visible: true)
+        #
+        #   # good
+        #   expect(page).to have_selector('.foo', visible: :all)
+        #
+        #   # good
+        #   expect(page).to have_selector('.foo', visible: :hidden)
+        #
+        #   # good
+        #   expect(page).to have_selector('.foo', visible: :visible)
+        #
+        class VisibilityMatcher < Cop
+          MSG = 'Use `:all`, `:hidden`, `:visible` instead of `true`, `false`.'
+
+          def_node_matcher :bad_method?, <<~PATTERN
+            (send nil? :have_selector ... (hash <(pair (sym :visible) {true false}) ...>))
```

Comment:
`:have_selector ...` should probably be something else, since the first, optional, parameter seems can be a symbol.
https://www.rubydoc.info/github/jnicklas/capybara/Capybara%2FNode%2FMatchers:has_selector%3F

This, if I understood correctly will match any amount of parameters before the hash, correct?

Reply:
[That's right](https://github.com/rubocop-hq/rubocop/blob/master/manual/node_pattern.md#-for-several-subsequent-nodes).

Reply:
I saw that, but I'm having a hard time changing this from `...` to something like `sym ?` by which I mean, "might take an optional first argument that is a symbol". But I guess the current `...` is also fine, even if it matches signatures that aren't allowed. WDYT?